autorest/utility_test.go

```diff
@@ -186,18 +186,52 @@ func TestEnsureStrings(t *testing.T) {
 		"string": "string",
 		"int":    42,
 		"nil":    nil,
+		"bytes":  []byte{255, 254, 253},
 	}
 	r := map[string]string{
 		"string": "string",
 		"int":    "42",
 		"nil":    "",
+		"bytes":  string([]byte{255, 254, 253}),
 	}
 	v := ensureValueStrings(m)
 	if !reflect.DeepEqual(v, r) {
 		t.Fatalf("autorest: ensureValueStrings returned %v\n", v)
 	}
 }
 
+func ExampleStringWithSeparator() {
```

Comment:
@garimakhulbe This and the following should be tests. Examples become part of the documentation. I don't believe we need to or should document these methods.